transcendence-app/src/game/stats/infrastructure/db/postgres/user-level.postgres.repository.ts

```diff
@@ -69,4 +73,53 @@ export class UserLevelPostgresRepository
       ? levelsData.map((level) => new UserLevelWithTimestamp(level))
       : null;
   }
+
+  private async getPaginatedUserGames(
+    username: string,
+    gameMode?: GameMode,
+  ): Promise<Game[] | null> {
+    let gamesData;
+    if (gameMode) {
+      gamesData = await makeQuery<GameData>(this.pool, {
+        text: `SELECT *
+               FROM ${table.GAME}
+               WHERE (${gameKeys.PLAYERONEUSERNAME} = $1
+                 OR ${gameKeys.PLAYERTWOUSERNAME} = $1)
+                 AND ${gameKeys.GAMEMODE} = $2;`,
+        values: [username, gameMode],
+      });
+    } else {
+      gamesData = await makeQuery<GameData>(this.pool, {
+        text: `SELECT *
+               FROM ${table.GAME}
+               WHERE ${gameKeys.PLAYERONEUSERNAME} = $1
+                 OR ${gameKeys.PLAYERTWOUSERNAME} = $1;`,
+        values: [username],
+      });
+    }
+    return gamesData ? gamesData.map((game) => new Game(game)) : null;
+  }
+
+  async getWinGameRatio(
+    username: string,
+    gameMode?: GameMode,
+  ): Promise<number> {
+    const games = await this.getPaginatedUserGames(username, gameMode);
+    if (!games) {
+      return 0;
+    }
+    let wonGames = 0;
+    games.forEach((game) => {
+      if (game.playerOneUsername === username) {
+        if (game.playerOneScore > game.playerTwoScore) {
+          wonGames++;
+        }
+      } else {
+        if (game.playerTwoScore > game.playerOneScore) {
+          wonGames++;
+        }
+      }
+    });
+    return wonGames / games.length;
```

Comment:
I guess it's possible to calculate this in pure postgrs sql? Either way, it's a bother, and I'm unwilling to do it, since it works just fine like this, and I'm not going to learn anything interesting from the effort.